core/tests/unittests/utils/test_utils.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+
+import numpy as np
+import pandas as pd
+
+from autogluon.core.utils import infer_problem_type
+from autogluon.core.constants import BINARY, MULTICLASS, MULTICLASS_UPPER_LIMIT, REGRESSION
+
+
+class TestInferProblemType(unittest.TestCase):
```

Comment:
Missing a test with mix of NaN and valid values

Reply:
Added